app/controllers/api/v1/games_controller.rb

```diff
@@ -23,15 +23,22 @@ def update
 
   def join
     game = Game.find(params['gameId'])
-    new_player = game.players.create(name: params['playerName'])
-    game.start
-    join_info = {
-      gameId: game.id,
-      playerId: new_player.id,
-      playerName: new_player.name,
-      gameStatus: "Game Started"
-    }
-    render json: join_info
+    max_players = 2
+    #change above variable to allow for more players
+    current_player = Player.find_by(name: params['playerName'])
+    if game.players.length >= max_players && current_player.nil?
+      render json: {error: "Game is full"}, status: 406
+    else
+      new_player = game.players.find_or_create_by(name: params['playerName'])
+      game.start if game.game_cards.length == 0 && game.players.length == max_players
+      join_info = {
+        gameId: game.id,
+        playerId: new_player.id,
+        playerName: new_player.name,
+        gameStatus: "Game Started",
```

Comment:
If we increase Player Count, we'll need to change this message to handle "Joined Lobby" and "Game Started" as well. But this looks really good, I like the dynamic logic!